etna/transforms/math/binary_operator.py

```diff
@@ -0,0 +1,166 @@
+from enum import Enum
+from typing import List
+from typing import Optional
+
+import pandas as pd
+
+from etna.datasets import TSDataset
+from etna.transforms.base import ReversibleTransform
+
+
+class BinaryOperator(str, Enum):
+    """Enum for mathematical operators from pandas."""
+
+    add = "+"
+    sub = "-"
+    mul = "*"
+    div = "/"
+    floordiv = "/="
+    mod = "%"
+    pow = "**"
+
+    eq = "=="
+    ne = "!="
+    le = "<="
+    lt = "<"
+    ge = ">="
+    gt = ">"
+
+    @classmethod
+    def _missing_(cls, value):
+        raise ValueError(f"Incorrect operand, literal {value} is unsupported")
+
+    def perform(self, df: pd.DataFrame, left_operand: str, right_operand: str, out_column: str) -> pd.DataFrame:
+        """Perform binary operation on passed dataframe."""
+        pandas_operator = getattr(pd.DataFrame, self.name)
+        df_left = df.loc[:, pd.IndexSlice[:, left_operand]].rename(columns={left_operand: out_column}, level="feature")
+        df_right = df.loc[:, pd.IndexSlice[:, right_operand]].rename(
+            columns={right_operand: out_column}, level="feature"
+        )
+        return pandas_operator(df_left, df_right)
+
+
+class BinaryOperationTransform(ReversibleTransform):
+    """Perform binary operation on the columns of dataset."""
+
+    def __init__(self, left_column: str, right_column: str, operator: str, out_column: Optional[str] = None):
+        """Create instance of BinaryOperationTransform.
+
+        Parameters
+        ----------
+        left_column:
+            Name of the left column
+        right_column:
+            Name of the right column
+        operator:
+            Operation to perform on the columns
+        out_column:
+            Resulting column name, if don't set, name will be `left_column operator right_column`.
+            If out_column is left_column or right_column, apply changes to the existing column out_column, else create new column.
+        """
+        inverse_logic = {"+": "-", "-": "+", "*": "/", "/": "*"}
+        super().__init__(required_features=[left_column, right_column, operator])
+        self.inplace_flag = (left_column == out_column) | (right_column == out_column)
+        self.left_column = left_column
+        self.right_column = right_column
+        if self.left_column == self.right_column:
+            raise ValueError("You should use LambdaTransform, when you perform operation only with one column")
+        self.operator = BinaryOperator(operator)
+        self.out_column = out_column if out_column is not None else self.left_column + self.operator + self.right_column
+
+        self._out_column_regressor: Optional[bool] = None
```

Comment:
Why do you choose name `_out_column_regressor`? In other transforms `_in_column_regressor` is used.